### PR TITLE
Added Golang Language Courses

### PR DESCRIPTION
**Pull request by Sushant-1999**
## What does this PR do?
Added Golang Language Courses by different Authors in Hindi under Golang Section.

## For resources
Youtube
### Why is this valuable (or not)?
In these courses, everything about golang language has been taught by these different authors. These cover syntax, creating Rest Apis in golang. Also, these videos teach about concurrency, channels, and go routines which are a very hot topics in interviews for golang. These videos will cover everything who is beginner golang developers.

### How do we know it's really free?
It is on Youtube

### For book lists, is it a book? For course lists, is it a course? etc.
These are free courses in Hindi on Youtube

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: courses/free-courses-hi.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ### Index
 
-* [Algorithms](#algorithms)
+* [Algorithms](#algols
```
Comment (eshellman) [inline courses/free-courses-hi.md:3]: typo

**Review by eshellman — CHANGES_REQUESTED** [review]
just fix the typo

File: courses/free-courses-hi.md (hunk 1 above)
Comment (Sushant-1999) [reply courses/free-courses-hi.md]: Done.
* [Algorithms](#algorithms)

Comment (Sushant-1999) [reply courses/free-courses-hi.md]: Changes are ready to merge now.

File: courses/free-courses-hi.md [hunk 2]
```diff
@@ -246,3 +247,11 @@
...
 
 * [Rust Complete Tutorial In Hindi](https://www.youtube.com/playlist?list=PLRuqvIc0eAmp8Lv6M4BKQWEinvCuqdVFP) - One Two Coding
+
+
+### Golang
+
+* [Go Lang Tutorials in HINDI](https://www.youtube.com/playlist?list=PL45_xGOyv4bk55CMmqH6S6vvnwKD8qifD) - Coder Singh
+* [Go programming (GoLang) Tutorial for Beginners in Hindi](https://www.youtube.com/playlist?list=PLEtkoO2np9szq1XDH1Mfr36Fg5EXh-z20) - Host Progrmming
+* [Go programming Tutorial for Beginners(Hindi)](https://www.youtube.com/playlist?list=PLgPJX9sVy92yu7If3I7GonlWA8YU1BuAk) - CS Geeks
+* [Let's go with golang (Hindi)](https://www.youtube.com/playlist?list=PLRAV69dS1uWQGDQoBYMZWKjzuhCaOnBpa) - Hitesh Choudhary
```
Comment (davorpa) [inline courses/free-courses-hi.md:257]: Using titles as same as provided by original source

```suggestion
* [Let's go with golang](https://www.youtube.com/playlist?list=PLRAV69dS1uWQGDQoBYMZWKjzuhCaOnBpa) - Hitesh Choudhary
```

Comment (Sushant-1999) [reply courses/free-courses-hi.md:257]: Any changes required from my side @davorpa ?

Comment (davorpa) [reply courses/free-courses-hi.md]: > Any changes required from my side @davorpa ?

Just commit the suggestion. Done!

**Review by davorpa — APPROVED** [review]

File: courses/free-courses-hi.md (hunk 1 above)
Comment (Sushant-1999) [reply courses/free-courses-hi.md]: Can you approve the changes @eshellman

**Review by davorpa — APPROVED** [review]

**Comment by Sushant-1999** [comment]
Conflicts has been resolved please merge @eshellman. This has been pending from long time.

**Comment by Sushant-1999** [comment]
Please merge so that I can contribute more.

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Comment by Sushant-1999** [comment]
My pleasure. Thanks !!!

**Pull request merged**